packages/next/server/next-dev-server.ts

```diff
@@ -214,6 +224,23 @@ export default class DevServer extends Server {
               match: getRouteMatcher(getRouteRegex(page)),
             }))
 
+          // we write a separate manifest with all pages for the client in
+          // dev mode so that we can match a page after a rewrite on the client
+          // before it has been built and is populated in the _buildManifest
+          fs.promises
```

Comment:
Is this purposely not awaited?

Reply:
Yeah since we're not inside of an async method the error is handled through an individual catch instead

Reply:
The method should be made async imo.

Reply:
Had a look at it and went for a different approach that does not require us to write any extra files 🚀